pkg/commonconfig/commonconfig.go

```diff
@@ -36,47 +37,49 @@ func (cc *CommonConfig) GetConfigFlags() genericclioptions.RESTClientGetter {
 	return *cc.ConfigFlags
 }
 
-func (cc *CommonConfig) GetProviderFromConfig() (string, error) {
-	config, err := cc.GetConfigFlags().ToRESTConfig()
-	if err != nil {
-		return "", microerror.Mask(err)
+func (cc *CommonConfig) GetInstallation(ctx context.Context, path, athenaUrl string) (*installation.Installation, error) {
+	if path == "" {
+		config, err := cc.GetConfigFlags().ToRESTConfig()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		path = config.Host
 	}
-	return cc.extractProviderFromUrl(config.Host), nil
+	if cc.installation == nil || cc.installation.SourcePath != path {
+		i, err := installation.New(ctx, path, athenaUrl)
+		cc.installation = i
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+	return cc.installation, nil
 }
 
-func (cc *CommonConfig) GetProviderFromInstallation(ctx context.Context, athenaUrl string, fallbackToConfig bool) (string, error) {
+func (cc *CommonConfig) GetProviderFromConfig(ctx context.Context, athenaUrl string) (string, error) {
```

Comment:
Getting provider info from the Installation data loaded from Athena, and falling back to API URL detection in case Athena is not available.